src/unix/getaddrinfo.c

```diff
@@ -200,3 +200,23 @@ void uv_freeaddrinfo(struct addrinfo* ai) {
   if (ai)
     freeaddrinfo(ai);
 }
+
+
+int uv_if_indextoname(unsigned int ifindex, char* ifname, size_t *sizep) {
+  char ifname_buf[UV_IF_NAMESIZE];
+  int required_len;
+  const int ifname_len = *sizep;
+
+  if (NULL == if_indextoname(ifindex, ifname_buf))
+    return -errno;
+
+  ifname_buf[sizeof(ifname_buf)-1] = '\0'; // To be _really_ sure
```

Comment:
I dropped the separate assignment, but also changed `strlen` to `strnlen` to avoid running over the buffer even in the case there was some obscure internal hiccup.